library/general/src/lib/cfa/nsswitch.rb

```diff
@@ -0,0 +1,273 @@
+      #
+      # @param file_handler [#read,#write] something able to read/write a string (like File)
+      # @return [Nsswitch] File with the already loaded content
+      def load(file_handler: Yast::TargetFile)
+        new(file_handler: file_handler).tap(&:load)
+      end
+    end
+
+    # Constructor
+    #
+    # @param file_handler [#read,#write] something able to read/write a string (like File)
+    #
+    # @see CFA::BaseModel#initialize
+    def initialize(file_handler: Yast::TargetFile)
+      super(AugeasParser.new("nsswitch.lns"), load_path, file_handler: file_handler)
+    end
+
+    # Loads the file content
+    #
+    # If the file does not exist, consider it as empty.
+    #
+    # @see CFA::BaseModel#load
+    def load
+      super
+      fix_collection_names(data)
+      @current_content = @parser.serialize(data)
+    rescue Errno::ENOENT # PATH does not exist yet
+      self.data = @parser.empty
+      @loaded = true
+    end
+
+    # Returns a list of defined databases
+    #
+    # @return [Array<String>] List of database names
+    def entries
+      databases.keys
+    end
+
+    # Service specifications for given database name
+    #
+    # @param db_name [String] the database name, e.g., "passwd" or "hosts"
+    #
+    # @return [Array<String>] database service specifications
+    def services_for(db_name)
+      databases[db_name]&.services
+    end
+
+    # Update (or create if it does not exist yet) the entry for given database name
+    #
+    #
+    # @note Current services, if any, are completely replaced
+    #
+    # @see #entry_for
+    # @see DatabaseEntry#services=
+    #
+    # @param db_name [String] the database name, e.g., "passwd" or "hosts"
+    # @param services [Array<String>] service specifications for the database
+    def update_entry(db_name, services)
+      database = entry_for(db_name)
+      database.services = services
+    end
+
+    # Delete entry for given database name
+    #
+    # @param db_name [String] the database db_name, e.g., "passwd" or "hosts"
+    def delete_entry(db_name)
+      data.delete(database_matcher(db_name))
+    end
+
+    # Writes the current content to /etc/nsswitch.conf
+    #
+    # @see CFA::BaseModel#save
+    def save
+      return false unless modified?
+
+      new_content = @parser.serialize(data)
+
+      @parser.file_name = write_path if @parser.respond_to?(:file_name=)
+      @file_handler.write(write_path, new_content)
+      @current_content = new_content
+
+      true
+    end
+
+    # Whether the content has changed
+    #
+    # @return [Boolean] true if content has been modified; false otherwise.
+    def modified?
+      @current_content != @parser.serialize(data)
+    end
+
+  private
+
+    # The path for loading the content
+    #
+    # @return [String] path to local configuration file if it exists; path to vendor one otherwise
+    def load_path
+      Yast::FileUtils.Exists(LOCAL_PATH) ? LOCAL_PATH : VENDOR_PATH
+    end
```

Comment:
well, I worry this is kind of tricky, because if you use different load_path and write path. Result is that write path will be always complete file. Which I am not sure if it is goal of that vendor config files, where it expect to have in local path only modifications and not complete config file, or am I wrong?

Reply:
> well, I worry this is kind of tricky, 

Yes, it is :(

> because if you use different load_path and write path. Result is that write path will be always complete file. 

Which is what we are looking for here.

> Which I am not sure if it is goal of that vendor config files, where it expect to have in local path only modifications and not complete config file, or am I wrong?

I think that's what we all expect, but is not true for this case (and probably others, but I don't know them yet). Maybe I should improve the PR description to explain it better, but basically there is a _variant_ of configuration files that are too complex and does not allow to override a single value in their local counterpart. So, once you want to change something but preserve the other values defined in the vendor file, you must make a full copy from /usr/etc/example.conf to /etc/example.conf file. If not, the default values will be used for entries not defined in the local config file.

This was tested by myself (see https://bugzilla.suse.com/show_bug.cgi?id=1173119#c2) and also documented in https://en.opensuse.org/openSUSE:Packaging_UsrEtc

I hope it is more clear now.

Reply:
yes, clear now. So please document it also in code. So basically vendor config acts as defaults and user one as modified.....I am not sure if this whole concept is good idea as it is hard for user to see if there are new/changed options than with old .rpmsave files, but it is different topic.